packages/manager/apps/pci-users/src/components/users/OpenStackDownloadModal.tsx

```diff
@@ -0,0 +1,185 @@
+import {
+  OsdsButton,
+  OsdsFormField,
+  OsdsIcon,
+  OsdsLink,
+  OsdsModal,
+  OsdsSelect,
+  OsdsSelectOption,
+  OsdsSpinner,
+  OsdsText,
+} from '@ovhcloud/ods-components/react';
+import {
+  ODS_BUTTON_VARIANT,
+  ODS_ICON_NAME,
+  ODS_ICON_SIZE,
+  ODS_SPINNER_SIZE,
+  OdsSelectValueChangeEventDetail,
+  OsdsSelectCustomEvent,
+} from '@ovhcloud/ods-components';
+import {
+  ODS_THEME_COLOR_INTENT,
+  ODS_THEME_TYPOGRAPHY_LEVEL,
+  ODS_THEME_TYPOGRAPHY_SIZE,
+} from '@ovhcloud/ods-common-theming';
+import { useTranslation } from 'react-i18next';
+import { OdsHTMLAnchorElementTarget } from '@ovhcloud/ods-common-core';
+import { useEnvironment } from '@ovh-ux/manager-react-shell-client';
+import { useEffect, useState } from 'react';
+import { useDownloadOpenStackConfig } from '@/hooks/useUser';
+import { OPENSTACK_GUIDE } from '@/download-openrc.constants';
+import { useAllRegions } from '@/hooks/useRegion';
+import { getMacroRegion, getOpenRcApiVersion } from '@/data/region';
+
+interface OpenStackDownloadModalProps {
+  projectId: string;
+  userId: string;
+  onClose: () => void;
+  onSuccess: (content: string) => void;
+  onError: (cause: Error) => void;
+}
+
+export default function OpenStackDownloadModal({
+  projectId,
+  userId,
+  onClose,
+  onSuccess,
+  onError,
+}: OpenStackDownloadModalProps) {
+  const { t } = useTranslation('common');
+  const { t: tRegion } = useTranslation('region');
+  const { ovhSubsidiary } = useEnvironment().getUser();
+  const region = useEnvironment().getRegion();
+  const [currentRegion, setCurrentRegion] = useState('');
+  const { data: regions, isLoading } = useAllRegions(`${projectId}`);
+  const [openRCApiVersion, setOpenRCApiVersion] = useState(2);
+  const {
+    download: downloadOpenStackConfig,
+    isLoading: isLoadingDownload,
+  } = useDownloadOpenStackConfig({
+    projectId: `${projectId}`,
+    userId,
+    onError: (err) => {
+      onClose();
+      onError(err);
+    },
+    onSuccess: (content: string) => {
+      onClose();
+      onSuccess(content);
+    },
+  });
+  const openStackGuideURL = OPENSTACK_GUIDE[region][ovhSubsidiary];
+  useEffect(() => {
+    if (regions) {
+      setOpenRCApiVersion(getOpenRcApiVersion(regions, region));
+      setCurrentRegion(regions[0].name);
+    }
+  }, [regions]);
+
+  return (
+    <>
+      <OsdsModal
+        headline={t('pci_projects_project_users_download-openrc_title')}
+        onOdsModalClose={onClose}
+      >
+        <slot name="content">
+          {!isLoading && !isLoadingDownload && (
+            <>
+              <OsdsText
+                slot="label"
+                level={ODS_THEME_TYPOGRAPHY_LEVEL.heading}
+                color={ODS_THEME_COLOR_INTENT.text}
+              >
+                {t('pci_projects_project_users_download-openrc_content')}
+              </OsdsText>
+
+              <OsdsLink
+                href={openStackGuideURL}
```

Comment:
can the guideURL be undefined ? if so maybe you need to hide the link, you can check the angularjs code to see how it was handheld